core/src/main/scala/io/finch/endpoint/attribute.scala

```diff
@@ -0,0 +1,124 @@
+package io.finch.endpoint
+
+import scala.reflect.ClassTag
+
+import cats.Id
+import cats.data.NonEmptyList
+import com.twitter.finagle.http.exp.{Multipart => FinagleMultipart}
+import com.twitter.util._
+import io.catbird.util.Rerunnable
+import io.finch._
+import io.finch.internal.Rs
+
+private abstract class Attribute[F[_], A](val name: String, val d: DecodeEntity[A], val tag: ClassTag[A])
```

Comment:
I'd vote we keep both `Attribute` and `FileUpload` under the same file, `multipart.scala`. We do that for params.